### ISSUE TITLE
incorrect wall intersection calculation?

### ISSUE THREAD

**Issue by jwrpalmer99** [manuelVo/foundryvtt-lichtgeschwindigkeit#6]
not sure why this level has real problems. There are no lights and the token has dimvision; the token vision should look like this (your module disabled)

<img src="https://i.imgur.com/YX9vVOR.png">

but with your module (I have disabled all other modules for this screenshot) it's like this:

<img src="https://i.imgur.com/EZO7cAM.png">

the walls are built like this:

<img src="https://i.imgur.com/7rKEIPp.png">

the raycasting debug is:

[https://pastebin.com/0B0iF9qc](https://pastebin.com/0B0iF9qc)

**Comment by manuelVo** [comment]
Thanks for the bugreport!

The map contains a lot oddities like walls that are fully overlapped by other walls and walls that have no size at all (meaning their start point is at the exact same location as their end point). The algorithm didn't expect the latter, which caused a glitch in the vision calculation. I've adjusted the algorithm to be able to deal with walls with no size.

If you want to see yourself what I mean, I've highlighted one of the sizeless walls in this screenshot. Remove the wall left, right and below the red circle from the map and the circled endpoint won't disapear. That is the sizeless wall.
![grafik](https://user-images.githubusercontent.com/1705157/122780760-6d4ffb00-d2af-11eb-8279-537efe8e83fa.png)
